utils/backoff_test.go

```diff
@@ -0,0 +1,64 @@
+package utils
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func TestProgressiveRetryShouldFail(t *testing.T) {
+	i := 0
+	operationError := errors.New("Operation Failed")
+	retryOperation := func() error {
+		i++
+		return operationError
+	}
+
+	err := ProgressiveRetry(retryOperation)
+
+	if err == nil || err != operationError {
+		t.Errorf("Expected %v, Got %v", operationError, err)
+	}
+	if i != 4 {
```

Comment:
Rather than testing `NextRetry` separately, could we find a way to verify that `ProgressRetry` actually waits between attempts? For example, instead of just incrementing `i`, perhaps we add the current time to a slice, and then just verify that the time between intervals is strictly increasing?

Reply:
See my comment about just using a table of durations, then no code to test either.